spec.html

```diff
@@ -357,10 +357,25 @@ <h1>Runtime Semantics: HostInitializeSyntheticRealm ( _realm_ )</h1>
 				operation used to inform the host of any newly created realms from
 				the Realm constructor. Its return value is not used, though it may
 				throw an exception. The idea of this hook is to initialize host
-				data structures related to the Realm, e.g., for module loading. It
-				is not expected that this hook would add properties to the Realm's
-				global object.
+				data structures related to the Realm, e.g., for module loading.
 			</p>
+			<p>
+				The host may use this hook to add properties to the Realm's global
+				object. Those properties must each be configurable to provide
+				platform capabilities with no authority to cause side effects such
+				as I/O or mutation of values that are shared across different
+				realms within the same host environment.
```

Comment:
> with no authority to cause side effects such as I/O or mutation of values that are shared across different realms

This is not enough for the SES requirement.

Imagine the host provide a `counter { [[Get]]: () => #counter, [[Set]]: (val) => #counter = val }` property on the Realm's global object, this provides a shared global state _within_ the Realm (not shared across different realms). It provides a communication channel in the different compartments within the same ream.

Reply:
I guess I read the text to cover that case--I see the code sample as a value shared across realms, at a high level. I guess we just need to iterate on the wording so that it's maximally clear.

Reply:
If `#counter` is a host-defined pre-realm internal state, it will be a mutable state within the Realm and passes the current spec text but violate the SES requirement.